### PR TITLE
 upscayl: init at 2.0.1 

### PR DESCRIPTION
**Pull request by TheZombie1999**
###### Description of changes

I added the upscaly appimage to nixpkgs.

This is my very first contribution so any feedback is very welcome.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rapenne-s** [comment]
nice! I'll review it. First problem I spot is one commit is used the name "upscaly" instead of upscayl, there is also the same problem in the package name.

**Comment by rapenne-s** [comment]
Your current version is missing a `meta` attribute with information such as homepage, license, platforms, maintainers etc...

**Comment by AmeerTaweel** [comment]
@TheZombie1999 any updates on this?

**Comment by onny** [comment]
Hey @TheZombie1999 thank you for this contribution! I would like to use this tool too. If you could squash the commits and change the commit title to `upscaly: init at X.X.X` I would be happy to review it :)

**Comment by TheZombie1999** [comment]
So i added the relevant meta attributes and updated the package to the newest version.

I am currently struggeling with the squashing because i have never done this before and don't know how deal with 
my unecessary mering of nixpkgs into my fork.

The other problem that i have is that i get some formatting errors. I tried to run nixpkgs-fmt on the two relevant files.

I seemed to have worked on the upscaly/default.nix file but not on the all-packges file . 

The problem is that it changes a lot of things that i have never touched.

**Review by Zahrun — CHANGES_REQUESTED** [review]

**Comment by Zahrun** [comment]
Please also rename the commit and pr to upscayl

File: pkgs/applications/video/upscayl/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{appimageTools, fetchurl}:
```
Comment (Zahrun) [inline pkgs/applications/video/upscayl/default.nix:1]: include lib
`{appimageTools, fetchurl, lib}:`

**Review by Zahrun — CHANGES_REQUESTED** [review]

**Comment by Zahrun** [comment]
Now this works, congratulations!
It would be nice to have a .desktop file so that upscayl also comes in the app menu.
You can see at https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/protonup-qt/default.nix how they did that.

**Comment by TheZombie1999** [comment]
I think I implemented the .desktop file correctly now.
But I don't use them my self, therefore I can't properly test it.
Can someone pls do that ?

**Comment by Zahrun** [comment]
I can see the .desktop entry in my app menu. However, it points to `upscayl` whereas the executable name is now `upscayl-2.0.1`. I guess you can just specify `name = upscayl` in appimageTools.wrapType2.

**Review by Zahrun — APPROVED** [review]
Excellent, thank you for your contribution!

**Comment by Icy-Thought** [comment]
Also tested package by upscayling several images and it appears to be working just fine!

File: pkgs/applications/video/upscayl/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      --replace 'Exec=AppRun --no-sandbox %U' 'Exec=${pname}'
+  '';
+
+  extraPkgs = pkgs: with pkgs; [ vulkan-headers vulkan-loader ];
+
+  meta = with lib; {
+    description = "Free and Open Source AI Image Upscaler";
+    homepage = "https://upscayl.github.io/";
+    maintainers = with maintainers; [ tim ];
+    platforms = platforms.linux ;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/upscayl/default.nix:33]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/applications/video/upscayl/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    url = "https://github.com/upscayl/upscayl/releases/download/v${version}/upscayl-${version}-linux.AppImage";
+    sha256 = "sha256-qsFYWUK5GwuiAaU7Y3tAM8Eea1G+R/MwoDSFUZvo/cM=";
+  };
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 rec {
+  inherit pname version src;
+
+  name = pname;
+
```
Comment (SuperSandro2000) [inline pkgs/applications/video/upscayl/default.nix:16]:
```suggestion
```
that would remove version

File: pkgs/applications/video/upscayl/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{appimageTools, fetchurl, lib}:
+
+let
```
Comment (SuperSandro2000) [inline pkgs/applications/video/upscayl/default.nix:3]:
```suggestion
let
```

File: pkgs/applications/video/upscayl/default.nix (hunk 3 above)
Comment (TheZombie1999) [reply pkgs/applications/video/upscayl/default.nix:16]: This is required so that the executable name is upscayl and not upscayl-2.0.1

Comment (SuperSandro2000) [reply pkgs/applications/video/upscayl/default.nix:16]: You want to rename the executable instead of overwriting name.

**Comment by takov751** [comment]
tested the application, it works well.

File: pkgs/applications/video/upscayl/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/applications/video/upscayl/default.nix:16]: This must be changed

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/upscayl/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "upscayl";
+  version = "2.0.1";
+  src = fetchurl {
+    url = "https://github.com/upscayl/upscayl/releases/download/v${version}/upscayl-${version}-linux.AppImage";
+    sha256 = "sha256-qsFYWUK5GwuiAaU7Y3tAM8Eea1G+R/MwoDSFUZvo/cM=";
```
Comment (drupol) [inline pkgs/applications/video/upscayl/default.nix:8]:
```suggestion
    hash = "sha256-qsFYWUK5GwuiAaU7Y3tAM8Eea1G+R/MwoDSFUZvo/cM=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by Icy-Thought** [comment]
Might as well deprecate this PR in favor of a newer one with the [latest version of Upscayl](https://github.com/upscayl/upscayl/releases/tag/v2.5.5).

**Comment by drupol** [comment]
Hi @TheZombie1999 ,

Do you plan to implement the changes?

If not, could you please close the PR please ?

Thanks

**Comment by Icy-Thought** [comment]
In the meantime I am going to push a new PR with the latest update.

**Comment by drupol** [comment]
Okay cool!

Send your PR and I'll close this one.

**Comment by drupol** [comment]
Thank you for closing it. If you still want to be maintainer of this derivation, please chime in https://github.com/NixOS/nixpkgs/pull/237494 !

**Pull request closed without merging**